proxy/tracing.go

```diff
@@ -112,26 +121,33 @@ func (t *proxyTracing) startFilterTracing(operation string, ctx *context) *filte
 	if t.disableFilterSpans {
 		return nil
 	}
-	span := tracing.CreateSpan(operation, ctx.request.Context(), t.tracer)
+
+	_, span := t.Start(ctx.request.Context(), operation)
 	ctx.parentSpan = span
 
 	return &filterTracing{span, t.logFilterLifecycleEvents}
 }
 
 func (t *filterTracing) finish() {
-	if t != nil {
-		t.span.Finish()
+	if t == nil {
+		return
 	}
+
+	t.span.End()
 }
 
 func (t *filterTracing) logStart(filterName string) {
-	if t != nil && t.logEvents {
```

Comment:
same as https://github.com/zalando/skipper/pull/2578/files#r1530916986 I would try to let all logic the same but replace the content of the block as necessary